saleor/graphql/product/bulk_mutations/category_bulk_delete.py

```diff
@@ -0,0 +1,29 @@
+import graphene
+
+from ....core.permissions import ProductPermissions
+from ....product import models
+from ....product.utils import delete_categories
+from ...core.mutations import ModelBulkDeleteMutation
+from ...core.types import NonNullList, ProductError
+from ...plugins.dataloaders import load_plugin_manager
+from ..types import Category
+
+
+class CategoryBulkDelete(ModelBulkDeleteMutation):
+    class Arguments:
+        ids = NonNullList(
+            graphene.ID, required=True, description="List of category IDs to delete."
+        )
+
+    class Meta:
+        description = "Deletes categories."
+        model = models.Category
+        object_type = Category
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        error_type_class = ProductError
+        error_type_field = "product_errors"
+
+    @classmethod
+    def bulk_action(cls, info, queryset):
+        manager = load_plugin_manager(info.context)
+        delete_categories(queryset.values_list("pk", flat=True), manager)
```

Comment:
This function triggers `CATEGORY_DELETED` and `PRODUCT_UPDATED` webhooks for all removed categories and each of their products. Is that what we want @patrys? Setting product's category to null also results in making this product unpublished within a channel.

Reply:
I don't think this should fire both. We must decide if the product/category association belongs to the product or the category.

As for unpublishing, we want to make categories optional (and possibly allow multiple categories to be assigned), so I don't think it should result in the product becoming unpublished. How does the API behave if a client requests the category field and no category is assigned? Do we return `null`? If so, we can make the category optional when this is merged.

Reply:
Not related to product bulk mutations, I will create a separate issue for that.